packages/db/src/search/StatsSearch.ts

```diff
@@ -0,0 +1,298 @@
+import { Knex } from 'knex';
+import omit from 'lodash/omit';
+import { getKnexClient } from '../connection';
+import { TableNames } from '../tables';
+import { DbQueryParameters } from '../types/search';
+import { BaseSearch, typeToTable } from './BaseSearch';
+
+type TotalSummary = {
+  count_errors: number,
+  count_collections: number,
+  count_granules: number,
+  avg_processing_time: number,
+};
+
+type Aggregate = {
+  count: string,
+  status?: string,
+  error?: string,
+  name?: string,
+};
+
+type Summary = {
+  dateFrom: string,
+  dateTo: string,
+  value: number,
+  aggregation: string,
+  unit: string,
+};
+
+type SummaryResult = {
+  errors: Summary,
+  granules: Summary,
+  collections: Summary,
+  processingTime: Summary,
+};
+
+type Meta = {
+  name: string,
+  count: number,
+  field: string,
+};
+
+type AggregateRes = {
+  key: string,
+  count: number,
+};
+
+type ApiAggregateResult = {
+  meta: Meta,
+  count: AggregateRes[]
+};
+
+const infixMapping = new Map([
+  ['granules', 'granule_id'],
+  ['collections', 'name'],
+  ['providers', 'name'],
+  ['executions', 'arn'],
+  ['pdrs', 'name'],
+]);
+
+/**
+ * A class to query postgres for the STATS and STATS/AGGREGATE endpoints
+ */
+class StatsSearch extends BaseSearch {
```

Comment:
suggest to add a new member, so you can use it in member functions.
readonly tableName: string;
constructor(event: QueryEvent, type: string) {
    super(event, type);
    this.tableName = typeToTable[this.type];
  }